### DISCUSSION TITLE
Dependency Dashboard Approval Notifications (& supply-chain attack concern)

### DISCUSSION THREAD

**Discussion by MIJOTHY-V2** [renovatebot/renovate#28213] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

Hi, we're evaluating Renovate suitability for our org (thus far impressions are very positive) and we're wondering if Renovate supports having a notification mechanism for dependency dashboard approval flow updates, analogous to how the default flow notifies repository maintainers of "update is available" via GH notifications from opening a PR.

For a bit of context - and to perhaps help me figure out if this is an XY problem as I've not come across this kind of risk mentioned in the docs - we are using the dependency dashboard approval flow because of concerns about supply-chain attacks due to unreviewed code running on CI with access to build-time sensitive information. To be specific about the attack we are concerned about, it takes the form of:

1. Application `app` depends on safe dependency `x` version `1.0.0`
2. Malicious actor introduces code in dependency `x` version `1.0.1` to exfiltrate sensitive build-time information (e.g. the GHA `secrets` context)
3. Automated dependency update tooling creates a PR on `app` repo to bump dependency `x` to version `1.0.1`.
4. PR creation triggers usual CI workflows, and dependency `x` on version `1.0.1` is able to exfiltrate sensitive build-time information 

My understanding is different tools/flows handle this in different ways:
* Renovate recommended default flow (`We do not recommend that you require approval for all updates.`) - no mitigation of this vulnerability, as the bot opens PRs which are able to trigger regular CI jobs
* Renovate `dependencyDashboardApproval` flow - full mitigation of this vulnerability, as repo maintainers have the opportunity to review updated dependency code prior to PR creation and CI jobs being run
* Dependabot flow - partial mitigation of this vulnerability, as CI jobs must have dedicated secrets supplied for dependabot. There's no opportunity for repo maintainers to review these PRs, but dedicated secrets allow for some restrictions of secrets/easier rotation etc.

Given the preference for using the `dependencyDashboardApproval` flow due to the above, an ideal situation for our devs would look like:
* Renovate detects new update for dependency is available
* Renovate updates dependency dashboard requesting approval for this update
* (missing piece) This dashboard update notifies repo maintainers of the change
* Repo maintainer reviews dependency code changes, and approves the update on the dashboard
* Renovate creates PR with dependency updates

There is some custom tooling we could write for the notification portion, but before doing that we thought it would be best to get Renovate maintainers' thoughts on both the perceived security issue, and the proposed flow for dependency updates, as well as if there are any existing options we've missed for setting this up. How have you seen other users handle this, if they do at all (if it's even a problem in the first place 😵‍💫)?

**Comment by rarkins** [comment, marked as answer]
This is not planned or feasible at this point.

From a CLI perspective (this repo), sending "Notifications" is not suitable. From a Mend app perspective, it's in theory possible, but we also don't have all the maintainer details. Also, sending _GitHub_ notifications is essentially possible because GitHub doesn't expose that flow via API.

I don't have any comments/opinions on the security aspect - it's up to you how you manage risk. Lowest security risk is to use no open source at all, or never upgrade once you have a program up and running.

**Reply by MIJOTHY-V2** [reply]
Thanks for the reply. The notifications explanation makes sense; we'll set up our own notification mechanism.

Re. the security considerations, I can appreciate the stance of not making recommendations as to how users should manage risk. Having said that, there are some recommendations and use-case descriptions in the docs which do cover different needs (e.g. the [use-cases section](https://docs.renovatebot.com/key-concepts/dashboard/#use-cases) and the recommendation for [not requiring approval for all updates](https://docs.renovatebot.com/key-concepts/dashboard/#require-approval-for-all-updates)). I would have found it very useful if there were some mention of supply-chain attack mitigation as a use-case in this section, because I imagine a major use-case of this flow _is_ to mitigate the risk of such attacks. If you think this would be worthwhile to document for future users, I can submit a PR to add a few sentences around this being another use-case in a neutral tone?
